pyspeckit/spectrum/units.py

```diff
@@ -400,6 +400,12 @@ def refX_units(self):
     def refX_units(self, value):
         log.warn("'refX_units' is deprecated; please use 'refX_unit'", DeprecationWarning)
         self.refX_unit = value
+    
+    def __getattr__(self, name):
+        if name == 'refX_unit':
+            return self.refX.unit
+        else:
+            object.__getattribute__(self, name)
```

Comment:
this falls in a maximum recursion loop since it calls again the same method. I still haven't found the way to do this, object.__getattribute__(self, name) didn't fall in the same loop but it didn't do nothing either. My problem is described here: http://stackoverflow.com/questions/12065570/how-to-use-default-property-descriptors-and-successfully-assign-from-init.
In the discussion they propose using an underscore for a "private" attribute or overwriting `__getattr`__ or `__setattr`__ or `__getattribute`__ sth proven not that easy. I suggest going with the underscore